packages/framework/src/Form/Admin/Product/Price/ProductPricesWithVatSelectType.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Form\Admin\Product\Price;
+
+use Override;
+use Shopsys\FrameworkBundle\Model\Pricing\Vat\VatFacade;
+use Shopsys\FrameworkBundle\Model\Product\ProductFacade;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Validator\Constraints;
+
+class ProductPricesWithVatSelectType extends AbstractType
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Vat\VatFacade $vatFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductFacade $productFacade
+     */
+    public function __construct(
+        protected readonly VatFacade $vatFacade,
+        protected readonly ProductFacade $productFacade,
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    #[Override]
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('vat', ChoiceType::class, [
+                'block_prefix' => 'product_prices_select_vat_input',
+                'required' => true,
+                'choices' => $this->vatFacade->getAllForDomainIncludingMarkedForDeletion($options['domain_id']),
+                'choice_label' => 'name',
+                'choice_value' => 'id',
+                'constraints' => [
+                    new Constraints\NotBlank(['message' => 'Please enter VAT rate']),
+                ],
+                'label' => t('VAT'),
+            ])
+            ->add(
+                'manualInputPricesByDomainAndPricingGroupId',
+                PricesWithCalculatedSellingPricesType::class,
+                [
+                    'label' => false,
+                    'required' => false,
+                    'domain_id' => $options['domain_id'],
+                    'selling_prices' => $options['selling_prices'],
+                ],
+            );
+    }
+
+    /**
+     * @param \Symfony\Component\OptionsResolver\OptionsResolver $resolver
+     */
+    #[Override]
+    public function configureOptions(OptionsResolver $resolver): void
```

Comment:
what about introducing a data object that would be set here as `data_class` - it would make the code more readable IMHO as you would be able to better navigate between this form type and `get/setProductCalculatedPricesGroup()` methods in `ProductData` object, what do you think? On the other hand, it would probably slightly complicate the code as you would need to create a factory for the new data class

Reply:
refactored, please take a look :)